app/controllers/concerns/requires_recent_mfa.rb

```diff
@@ -0,0 +1,24 @@
+module RequiresRecentMfa
+  class MissingMfaMethod < StandardError; end
+
+  extend ActiveSupport::Concern
+
+  def has_done_mfa_recently?
+    return true unless MultiFactorAuth.is_enabled?
+
+    session[:has_done_mfa]
```

Comment:
This is a boolean right?

Reply:
Yes, it's set by `do_sign_in` and `RedoMfaPhoneController`